src/commands/Minion/alch.ts

```diff
@@ -0,0 +1,153 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+import { BotCommand } from '../../lib/BotCommand';
+import getOSItem from '../../lib/util/getOSItem';
+import { Util } from 'oldschooljs';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import resolveItems from '../../lib/util/resolveItems';
+import hasItemEquipped from '../../lib/gear/functions/hasItemEquipped';
+import {
+	bankHasAllItemsFromBank,
+	formatDuration,
+	removeBankFromBank,
+	resolveNameBank
+} from '../../lib/util';
+import createReadableItemListFromBank from '../../lib/util/createReadableItemListFromTuple';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+import { AlchingActivityTaskOptions } from '../../lib/types/minions';
+import { Activity, Tasks, Time } from '../../lib/constants';
+import { rand } from '../../util';
+
+const options = {
+	max: 1,
+	time: 10_000,
+	errors: ['time']
+};
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			cooldown: 1,
+			usage: '<quantity:int{1}|itemName:...string> [itemName:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	async run(msg: KlasaMessage, [quantity, itemName]: [number | null | string, string]) {
+		if (typeof quantity === 'string') {
+			itemName = quantity;
+			quantity = null;
+		}
+
+		const { id: itemId, highalch, name } = getOSItem(itemName);
+		const numOfItems = await msg.author.numberOfItemInBank(itemId);
+		// 5 tick action
+		const timePerAlch = Time.Second * 3;
+
+		if (numOfItems === 0) throw `'You don't have any of this item to alch'`;
+
+		if (!quantity) {
+			quantity = numOfItems;
+		}
+
+		const maxCasts = Math.min(Math.floor(msg.author.maxTripLength / timePerAlch), numOfItems);
+
+		if (quantity * timePerAlch > msg.author.maxTripLength) {
+			throw `The max number of alchs you can do is ${maxCasts}!`;
+		}
+
+		const duration = quantity * timePerAlch;
+		let fireRuneCost = quantity * 5;
+
+		const unlimitedFireRuneProviders = resolveItems([
```

Comment:
define this outside of the `run` function so it doesnt have to be remade every execution